usr/src/uts/common/vm/vm_page.c

```diff
@@ -5701,15 +5702,22 @@ int
 page_reclaim_mem(pgcnt_t npages, pgcnt_t epages, int adjust)
 {
 	int	i = 0;
+	int	i_nofree = 0;
 	int	ret = 0;
 	pgcnt_t	deficit;
-	pgcnt_t old_availrmem;
+	pgcnt_t old_availrmem = 0;
 
 	mutex_enter(&freemem_lock);
-	old_availrmem = availrmem - 1;
-	while ((availrmem < tune.t_minarmem + npages + epages) &&
-	    (old_availrmem < availrmem) && (i++ < MAX_CNT)) {
-		old_availrmem = availrmem;
+	while (availrmem < tune.t_minarmem + npages + epages &&
+	    i++ < reclaim_maxcnt) {
+		/* ensure we made some progress in the last few iterations */
+		if (old_availrmem < availrmem) {
+			old_availrmem = availrmem;
+			i_nofree = 0;
+		} else if (i_nofree++ > reclaim_nofree_maxcnt) {
```

Comment:
Now that we retry harder, could this potentially have undesired side effect of reclaiming more than needed ? Or the deficit calculation below and the availrmem check in the while condition probably takes care of that?